PurchaseController/Classes/ProductsInfoController.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  ProductsInfoController.swift
+//  PurchaseController
+//
+//  Created by Igor Kulik on 7/15/19.
+//
+
+import Foundation
+import StoreKit
+
+typealias RetrievedProductsInfo = (products: Set<SKProduct>, invalidProductIdentifiers: [String])
+
+protocol ProductsInfoObserver: class {
+    var onRetrieve: ((_ productsInfo: RetrievedProductsInfo) -> Void)? { get set }
+    var onError: ((_ error: Error) -> Void)? { get set }
+}
+
+final class PCProductsInfoController: NSObject {
+    
+    // MARK: - Properties
+
+    private weak var observer: ProductsInfoObserver?
+    private var productsRequest: SKProductsRequest?
```

Comment:
can it be weak?

Reply:
No, as per our latest discussion, same case also included in 
https://github.com/dashdevs/PurchaseController/pull/8